providers/supermicro/supermicro.go

```diff
@@ -111,6 +111,12 @@ func (s *supermicro) ListAvailableUpdates(context.Context, *model.UpdateOptions)
 	return nil, nil
 }
 
+// UpdateRequirements returns requirements to be met before and after a firmware install,
+// the caller may use the information to determine if a powercycle, reconfiguration or other actions are required on the component.
+func (s *supermicro) UpdateRequirements(_ context.Context, componentSlug, componentVendor, componentModel string) (model.UpdateRequirements, error) {
+	return actions.UpdateRequirements(componentSlug, componentVendor, componentModel)
```

Comment:
Why not just call this function directly from the caller of the supermicro provider?

Reply:
The way clients currently returned a DeviceManager interface by ironlib - https://github.com/metal-toolbox/ironlib/blob/968097b876d07b81ef04da47c3d004a785452066/actions/interface.go#L13

Which determines the provider. We could have this work differently if the client knows what hardware its running on before hand through hints.